integration/gateway/gateway_suite_test.go

```diff
@@ -0,0 +1,169 @@
+)
+
+var _ = SynchronizedBeforeSuite(func() []byte {
+	buildServer = nwo.NewBuildServer()
+	buildServer.Serve()
+
+	components = buildServer.Components()
+	payload, err := json.Marshal(components)
+	Expect(err).NotTo(HaveOccurred())
+
+	return payload
+}, func(payload []byte) {
+	err := json.Unmarshal(payload, &components)
+	Expect(err).NotTo(HaveOccurred())
+
+	flogging.SetWriter(GinkgoWriter)
+})
+
+var _ = SynchronizedAfterSuite(func() {
+}, func() {
+	buildServer.Shutdown()
+})
+
+func StartPort() int {
+	return integration.GatewayBasePort.StartPortForNode()
+}
+
+func grpcClient(tlsRootCertFile string) *comm.GRPCClient {
+	caPEM, res := ioutil.ReadFile(tlsRootCertFile)
+	Expect(res).To(BeNil())
+
+	gClient, err := comm.NewGRPCClient(comm.ClientConfig{
+		Timeout: 3 * time.Second,
+		KaOpts:  comm.DefaultKeepaliveOptions,
+		SecOpts: comm.SecureOptions{
+			UseTLS:        true,
+			ServerRootCAs: [][]byte{caPEM},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(gClient).NotTo(BeNil())
+
+	return gClient
+}
+
+func GatewayClient(address, tlsRootCertFile string) gateway.GatewayClient {
+	grpcClient := grpcClient(tlsRootCertFile)
+	conn, err := grpcClient.NewConnection(address)
+	Expect(err).NotTo(HaveOccurred())
+	gatewayClient := gateway.NewGatewayClient(conn)
+
+	return gatewayClient
+}
+
+func NewProposedTransaction(signingIdentity msp.SigningIdentity, channelName, chaincodeName, transactionName string, args ...[]byte) *gateway.ProposedTransaction {
+	proposal, txnID := newProposalProto(signingIdentity, channelName, chaincodeName, transactionName, args...)
+	signedProposal, err := protoutil.GetSignedProposal(proposal, signingIdentity)
+	Expect(err).NotTo(HaveOccurred())
+
+	txn := &gateway.ProposedTransaction{
+		Proposal:  signedProposal,
+		TxId:      txnID,
+		ChannelId: channelName,
+	}
+
+	return txn
+}
+
+func newProposalProto(signingIdentity msp.SigningIdentity, channelName, chaincodeName, transactionName string, args ...[]byte) (*peer.Proposal, string) {
+	creator, err := signingIdentity.Serialize()
+	Expect(err).NotTo(HaveOccurred())
+
+	invocationSpec := &peer.ChaincodeInvocationSpec{
+		ChaincodeSpec: &peer.ChaincodeSpec{
+			Type:        peer.ChaincodeSpec_NODE,
+			ChaincodeId: &peer.ChaincodeID{Name: chaincodeName},
+			Input:       &peer.ChaincodeInput{Args: chaincodeArgs(transactionName, args...)},
+		},
+	}
+
+	result, transactionID, err := protoutil.CreateChaincodeProposal(
+		common.HeaderType_ENDORSER_TRANSACTION,
+		channelName,
+		invocationSpec,
+		creator,
+	)
+	Expect(err).NotTo(HaveOccurred())
+
+	return result, transactionID
+}
+
+func chaincodeArgs(transactionName string, args ...[]byte) [][]byte {
+	result := make([][]byte, len(args)+1)
+
+	result[0] = []byte(transactionName)
+	copy(result[1:], args)
+
+	return result
+}
+
+// TODO not the only suite doing this - move this to nwo?
+func GetSigningIdentity(network *nwo.Network, org, user, mspID, mspType string) msp.SigningIdentity {
```

Comment:
This doesn't make sense to me. If we need key material for a user or admin, the network provides functions to get the keys and certs. If we need the key for an orderer or a peer, I'd ask why, but we should not need an MSP to sign a transaction.

```
// OrdererUserKey returns the path to the private key for the specified user in
// the orderer organization.
func (n *Network) OrdererUserKey(o *Orderer, user string) string { ... }
```

```
// PeerUserKey returns the path to the private key for the specified user in
// the peer organization.
func (n *Network) PeerUserKey(p *Peer, user string) string { ... }
```